nibabel/nifti1.py

```diff
@@ -723,6 +723,11 @@ def set_data_shape(self, shape):
         If ``ndims == len(shape)`` then we set zooms for dimensions higher than
         ``ndims`` to 1.0
 
+        Nifti1 images can have up to seven dimensions. For Nifti surface files,
```

Comment:
How about the following documentation output?

https://qnl-linux1.bu.edu/~effigies/nibabel/reference/nibabel.nifti1.html#nibabel.nifti1.Nifti1Header.set_data_shape